test/cql-pytest/test_restrictions.py

```diff
@@ -33,3 +38,36 @@ def test_multiple_eq_restrictions_on_pk(cql, table1, scylla_only):
     assert [] == list(cql.execute(f'SELECT * FROM {table1} WHERE a = {p} AND a = {p2}'))
     assert [] == list(cql.execute(f'SELECT * FROM {table1} WHERE a = {p2} AND a = {p}'))
     assert [(p, 3)] == list(cql.execute(f'SELECT * FROM {table1} WHERE a = {p} AND a = {p}'))
+
+# Cassandra also doesn't allow restricting the same clustering-key column
+# more than once, producing all sorts of different error messages depending
+# if the operator is an equality or comparison and which order (see examples
+# in issue #12472). Scylla allows all these combination, and this test
+# verifies that they are implemented correctly:
+def test_multiple_restrictions_on_ck(cql, table2, scylla_only):
+    p = unique_key_int()
+    cql.execute(f'INSERT INTO {table2} (a, b) VALUES ({p}, 3)')
+    # b = ? and b = ?
+    assert [] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b = 3 AND b = 4'))
+    assert [] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b = 4 AND b = 3'))
+    assert [(p, 3)] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b = 3 AND b = 3'))
+    # b = ? and b > ?
+    assert [] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b = 3 AND b > 4'))
+    assert [] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b > 4 AND b = 3'))
+    assert [(p, 3)] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b = 3 AND b > 2'))
+    assert [(p, 3)] == list(cql.execute(f'SELECT * FROM {table2} WHERE a = {p} AND b > 2 AND b = 3'))
```

Comment:
This test would be more interesting if there was another row which had  a `b` larger than two. I think you should add a few more rows to the test table.

Reply:
I'll do this, but actually the point of my test wasn't to verify that filtering works correctly in general (we have a gazillion other tests for that) but that we don't do something stupid like having the second restriction for column b override the first restriction (e.g., that we don't put the restrictions in a hash table or something like that). This is also why I tested the restrictions in both orders - `b = 3 AND b > 2` and `b > 2 AND b =3` 

But it can't hurt to have more rows, so I'll send a v2.